src/stemmabench/variant_analyzer.py

```diff
@@ -0,0 +1,703 @@
+                Defaults to "en".
+        """
+        # alignment table
+        self.table = table
+        self.variant_locations: List[bool] = [col.variant for col in table.columns] 
+        # language
+        if language not in SUPPORTED_LANGUAGES:
+            logger.critical(f"Unknown language {language}.")
+            raise ValueError(f"Unknown language {language}.")
+        self.language = language
+        self.disable_synonym = disable_synonym
+
+    @property
+    def witness_names(self) -> List[str]:
+        """Get the list of witness names from an AlignmentTable.
+
+        Returns:
+            List[str]: The list of witness names extracted from the table 
+                rows.
+        """
+        return [row.header for row in self.table.rows]
+    
+    @property
+    def array(self) -> np.ndarray[np.ndarray]:
+        """Get an array representing an input Collatex AlignmentTable. 
+
+        Returns:
+            np.ndarray: A 2D numpy array representing the alignment table.
+                - Rows are witnesses.
+                - Columns are readings.
+        """
+        return self.alignment_table_to_numpy(self.table)
+    
+    #------------------------------------------------------------------------
+    # ------------ UTILITY METHODS ------------------------------------------
+    #------------------------------------------------------------------------
+    @staticmethod
+    def _get_token_strings(tokens_list: List[str | None], missing_reading:str="-"
+        ) -> List[str]:
+        """Extract token strings from a list of tokens or None values.
+        """
+        return [token_string.strip() if token_string else missing_reading 
+                    for token_string in tokens_list]
+
+    ### CLASS METHODS
+    @classmethod
+    def alignment_table_to_numpy(cls, table: AlignmentTable) -> np.ndarray[object]:
+        """
+        Convert an AlignmentTable to a numpy array.
+
+        Args:
+            table (AlignmentTable): The AlignmentTable to be converted.
+
+        Returns:
+            np.ndarray: A 2-D numpy array representing the alignment table.
+                - Rows (first dimension) correspond to the witnesses.
+                - Columns (second dimension) correspond to the readings.
+        """
+        # Create a dictionary to store witness data.
+        witness_data = {row.header: cls._get_token_strings(
+            row.to_list_of_strings()) for row in table.rows}
+        # Get the list of row names (witness names).
+        row_names = list(witness_data.keys())
+        # Create an empty numpy array with the same shape as the alignment table.
+        alignment_array = np.empty(shape=(len(table.rows), len(table.columns)), 
+                                   dtype=object)
+        # Fill the numpy array with the token strings lists from the witness data.
+        alignment_array = np.array([witness_data[name] for name in row_names], 
```

Comment:
why do you need to create an empty one before ? I don't think you're using it afterwards

Reply:
- [x] remove empty array initialization